File: frontend/lib/models/task.dart

class Task {
  // each task will have a text, completed, time made and time completed
  String text;
  DateTime dateCreated;
  String id;

  bool isCompleted;

  Task({
    required this.id,
    required this.text,
    required this.dateCreated,
    required this.isCompleted,
  });

  //create task object from a json
  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json["_id"],
      text: json["text"],
      isCompleted: json["isCompleted"],
      dateCreated: DateTime.parse(json["dateCreated"]),
    );
  }

  //convert task object to a json
  Map<String, dynamic> toJson() {
    return {
      "_id": id,
      "text": text,
      "isCompleted": isCompleted,
      "dateCreated": dateCreated.toIso8601String(),
    };
  }
}

File: backend/models/Task.js
const mongoose = require("mongoose");

const TaskSchema = new mongoose.Schema({
	text: {
		type: String,
		required: true,
	},
	dateCreated: {
		type: Date,
		default: Date.now,
		required: true,
	},

	isCompleted: {
		type: Boolean,
		default: false,
		required: true,
	},
});

const Task = mongoose.model("Task", TaskSchema);

module.exports = Task;
